conjure-generator-common/src/main/java/com/palantir/conjure/CamelCasePattern.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * (c) Copyright 2021 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.conjure;
+
+import com.google.errorprone.annotations.Immutable;
+
+@Immutable
+public final class CamelCasePattern implements SimplifiedPattern {
+    @Override
+    public String pattern() {
+        return "^[a-z]([A-Z]{1,2}[a-z0-9]|[a-z0-9])*[A-Z]?$";
+    }
+
+    @Override
+    @SuppressWarnings("checkstyle:CyclomaticComplexity")
+    public boolean matches(String value) {
```

Comment:
I'll need to set aside some time to review these carefully.

Reply:
@raiju when I hand-rolled a parser for sls-versions, I just put a 'jqwik' property-based testing to effectively fuzz the thing. https://github.com/palantir/sls-version-java/pull/455/files#diff-d90462e86f4e5e19c0df1b4545c7f9dcd7b440d389a391db7eca402d2f39b520R42

Helped catch the odd edge case for me 🤷

Reply:
In the glorious habit of "not invented here", I added a little fuzzing test for each parser, comparing results w/ the regex: https://github.com/palantir/conjure/pull/999/files#diff-85b295230e0ec833cca21a2ab5b43c5015ec4436fbfacb263949786e4df0a45cR54

Reply:
Moved it to the annotation-based one, and of course the limits of my little fuzzer came to light: my package parser was wrong for single-char packages, but my fuzzer only generated constant-length strings 🤦